File: src/com/anathema_roguelike/entities/characters/perks/actions/targetingstrategies/constraints/TargetConstraint.java
package com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.constraints;

import com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.Targetable;

public interface TargetConstraint<T extends Targetable, A> {
	boolean apply(T target, A arg);
}

File: src/com/anathema_roguelike/entities/characters/perks/actions/targetingstrategies/constraints/TargetConstraint.scala
package com.anathema_roguelike
package entities.characters.perks.actions.targetingstrategies.constraints

import com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.Targetable

trait TargetConstraint[TargetType <: Targetable, ArgType] {
  def apply(target: TargetType, arg: ArgType): Boolean
}
